core/tee/tee_rpmb_fs.c

```diff
@@ -505,6 +512,25 @@ static TEE_Result read_fat(struct rpmb_file_handle *fh, tee_mm_pool_t *p)
 	return res;
 }
 
+#ifdef CFG_ENC_FS
+static TEE_Result generate_fek_and_nonce(struct rpmb_fat_entry *fe)
+{
+	TEE_Result res;
+
+	res = crypto_ops.prng.read(fe->fek, sizeof(fe->fek));
+	if (res != TEE_SUCCESS)
+		return res;
+
+	res = crypto_ops.prng.read(fe->nonce, sizeof(fe->nonce));
+	return res;
+}
+#else
+static TEE_Result generate_fek_and_nonce(struct rpmb_fat_entry *fe __unused)
+{
+	return TEE_SUCCESS;
```

Comment:
Well `rpmb_fat_entry` struct is supposed to be zero-initialized at this point, but it won't hurt to clear `fek` again. As for `nonce` it will be removed as I said above. I will update this function.